src/engine.rs

```diff
@@ -648,3 +652,42 @@ impl<'a> DownloadsMapped<'a> {
         Ok(slice.get_mapped_range())
     }
 }
+
+const SIZE_CLASS_BITS: u32 = 1;
+
+impl ResourcePool {
+    /// Get a buffer from the pool or create one.
+    fn get_buf(&mut self, size: u64, usage: BufferUsages, device: &Device) -> Buffer {
+        let rounded_size = Self::size_class(size, SIZE_CLASS_BITS);
+        if let Some(buf_vec) = self.bufs.get_mut(&(rounded_size, usage)) {
+            if let Some(buf) = buf_vec.pop() {
+                return buf;
+            }
+        }
+        device.create_buffer(&wgpu::BufferDescriptor {
+            label: None,
+            size: rounded_size,
+            usage,
+            mapped_at_creation: false,
+        })
+    }
+
+    fn reap_bindmap(&mut self, bind_map: BindMap) {
+        for (_id, buf) in bind_map.buf_map {
+            let size = buf.size();
+            let usage = buf.usage();
+            self.bufs.entry((size, usage)).or_default().push(buf);
+        }
+    }
+
+    /// Quantize a size up to the nearest size class.
+    fn size_class(x: u64, bits: u32) -> u64 {
```

Comment:
For very small values of `x`, this is very stingy, likely requiring buffer recreation for small variations in size.

Now it is true that most small buffers will be fixed size anyway, for uniforms and such.
So it's probably that this doesn't matter.

Reply:
Yeah again the thinking here is to use simple logic, and fine-tuned based on measurements. There's no evidence yet that this kind of change would make any actual difference.